Extend/Controller/OrderController.php

```diff
@@ -286,4 +377,118 @@ private function _isFinalizeOrderSuccessful($iSuccess)
     {
         return $iSuccess === Order::ORDER_STATE_MAILINGERROR || $iSuccess === Order::ORDER_STATE_OK;
     }
+
+    /**
+     *
+     * Returns the parameters used to render the credit card form
+     *
+     * @return string
+     * @throws \Exception
+     */
+    public function getInitCreditCardFormJavaScript(): string
+    {
+
+        /**
+         * @var $oPaymentGateway PaymentGateway
+         */
+        $oPaymentGateway = oxNew(PaymentGateway::class);
+
+        /**
+         * @var $oBasket Basket
+         */
+        $oBasket = $this->getBasket();
+
+        /**
+         * @var $oOrder Order
+         */
+        $oOrder = oxNew(Order::class);
+        $oOrder->createTemp($oBasket, $this->getUser());
+
+        $oTransaction = $oPaymentGateway->createTransaction($oBasket, $oOrder);
+
+        $oTransaction->setAmount(new Amount(
+            $oBasket->getPrice()->getBruttoPrice(),
+            $oBasket->getBasketCurrency()->name
+        ));
+        $oTransaction->setConfig($this->_getCreditCardPaymentMethodConfig()->get(CreditCardTransaction::NAME));
+
+        $oSession = $this->getSession();
+        $sSid = $oSession->sid(true);
+        if ($sSid != '') {
+            $sSid = '&' . $sSid;
+        }
+
+        $sModuleToken = PaymentGateway::getModuleToken($oSession);
+        $oTransaction->setTermUrl(
+            Registry::getConfig()->getCurrentShopUrl() . "index.php?cl=order&" . $sModuleToken . $sSid
+        );
+
+        $oTransactionService = $this->_getTransactionService();
+
+        $oPayment = oxNew(Payment::class);
+        $oPayment->load($oBasket->getPaymentId());
+
+        return "ModuleCreditCardForm.init(" .
```

Comment:
Comment that this gets loaded in the template and outputs the data to be used in javascript